plugin/upstream_generator/bq_upstream_generator.go

```diff
@@ -95,18 +102,24 @@ func NewBQUpstreamGenerator(logger log.Logger, parserFunc ParserFunc, bqExtracto
 	if bqExtractorFunc == nil {
 		me.Append(fmt.Errorf("bqExtractorFunc is nil"))
 	}
-	if evaluatorFunc == nil {
-		me.Append(fmt.Errorf("evaluatorFunc is nil"))
+	sanitizedEvaluatorFuncs := []EvalAssetFunc{}
+	for _, evaluatorFunc := range evaluatorFuncs {
```

Comment:
assuming the argument always correct is dangerous. we should control the correctness of the arguments in constructor, so that we could safely call this anywhere within this object's methods

Reply:
yes, obviously that's also included in my consideration. that's why I didn't request to change it, but more on stating that it is "okay" to not include it. I never say that it is not dangerous in any way. I'm saying that with such assumption, we don't really need such validation. the reason for saying is that many in Optimus implementations we have such assumptions.